gwnr/waveform/esigma_utils.py

```diff
@@ -468,83 +675,132 @@ def get_imr_esigma_modes(
+                                  ['x', 'e', 'l', 'phi', 'phidot', 'r', 'rdot'].
+                                     Note that these are available only for the
+                                     inspiral portion of the waveform!
+        failsafe                  -- If True, we make reasonable choices for the
+                                     user, if the inputs to this method lead
+                                     into exceptions.
+        verbose                   -- Verbosity level. Available values are: 0, 1, 2
 
     Returns:
     --------
-        modes_imr         -- Dictionary of IMR GW modes PyCBC TimeSeries
+        modes_imr         -- Dictionary of IMR GW modes (PyCBC TimeSeries)
         orbital_var_dict  -- Dictionary of evolution of orbital elements.
-                             Returned only if "return_orbital_params" is specified
-        retval            -- Hybridization related data.
-                             Returned only if "return_hybridization_info" is True
+                             Returned only if the flag `return_orbital_params`
+                             is set
+        retval            -- Hybridization related data. Returned only if the
+                             flag `return_hybridization_info` is set
     """
     if not hasattr(ls, merger_ringdown_approximant):
         raise IOError(
             """We cannot generate individual modes for {merger_ringdown_approximant}.
                       Try one of: [NRSur7dq4, SEOBNRv4PHM]"""
         )
+    if (mean_anomaly is None) and (coa_phase is None):
+        raise IOError(
+            f"""Please specify one of the phase angles, either of
+                      `mean_anomaly` or `coa_phase`."""
+        )
+    if hybridize_aligning_merger_to_inspiral and (coa_phase is None):
+        raise IOError(
+            f"""If you want to attach ESIGMA inspiral to merger, by
+                      phase shifting merger to inspiral, please specify the
+                      phase angle `coa_phase`"""
+        )
+    if (not hybridize_aligning_merger_to_inspiral) and (mean_anomaly is None):
+        raise IOError(
+            f"""If you want to attach ESIGMA inspiral to merger, by
+                      phase shifting inspiral to merger, please specify the
+                      phase angle `mean_anomaly`"""
+        )
+    if mean_anomaly is None:
+        mean_anomaly = 0
+    if coa_phase is None:
+        coa_phase = 0
 
-    if return_orbital_params is True:
+    if return_orbital_params == True:
         return_orbital_params = ["x", "e", "l", "phi", "phidot", "r", "rdot"]
-
-    if isinstance(return_orbital_params, list):
+    elif isinstance(return_orbital_params, list):
         return_orbital_params_user = return_orbital_params.copy()
-    else:
+        return_orbital_params_user = set(return_orbital_params_user).intersection(
+            set(["x", "e", "l", "phi", "phidot", "r", "rdot"])
```

Comment:
This is a bit dangerous because if the user does not enter a valid orbital variable name, they will not get any error now (via the code in `get_inspiral_esigma_modes()` function). I think an error is better than just mysteriously not getting anything for a wrongly typed variable name. The user should know which variables are available and which are not.

Reply:
This is a good point, but I think a warning will be sufficient instead of an error here. It would be good to return the primary objects of this function, ie the IMR modes, despite typos in the orbital variable names. I have added a warning print here to make this very very explicit.